moshi/src/main/java/com/squareup/moshi/internal/Util.kt

```diff
@@ -499,6 +499,19 @@ public fun <T> Class<T>.boxIfPrimitive(): Class<T> {
   return wrapped ?: this
 }
 
+internal inline fun <T : Any> requireNull(value: T?, lazyMessage: (T) -> Any) {
```

Comment:
Pick a PR you want to have this addition in and let's get that done first, otherwise just going to have conflicts here. Same comments apply as on the other one

Reply:
Ok, I'll rebase this once the other PR is merged